lib/ui/recommendation.js

```diff
@@ -324,13 +378,13 @@ Recommendation.prototype = {
       // at the domain.
       // For wikipedia results, use nsIURI.equals() to check for URL equality.
       // If a duplicate is found, immediately exit.
-      if (type === 'tld' && this.eTLD.getBaseDomain(resultUri) === this.eTLD.getBaseDomain(recommendationUri) ||
-          type === 'wikipedia' && resultUri.equals(recommendationUri)) {
+      if (data.type === 'tld' && this.eTLD.getBaseDomain(resultUri) === this.eTLD.getBaseDomain(recommendationUri) ||
+          data.type === 'wikipedia' && resultUri.equals(recommendationUri)) {
         return;
```

Comment:
Need to add the movie type here. Nice, simple explanation for why dupe checking fails to work ;-)